lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -541,6 +541,85 @@ static void getCellAsmResultNames(OpAsmSetValueNameFn setNameFn, Operation *op,
     setNameFn(op->getResult(i), prefix + portNames[i].str());
 }
 
+//===----------------------------------------------------------------------===//
+// AssignOp
+//===----------------------------------------------------------------------===//
+
+/// Returns whether the given value is a port of a component, which are
+/// defined as Block Arguments.
+static bool isComponentPort(Value value) {
+  return value.getImpl()->getKind() == detail::ValueImpl::Kind::BlockArgument;
+}
+
+/// Verifies the given value of the AssignOp if it is a component port. The
+/// `isDestination` boolean is used to distinguish whether the value is a source
+/// or a destination.
+static LogicalResult verifyAssignOpWithComponentPort(AssignOp op,
+                                                     bool isDestination) {
+  Value value = isDestination ? op.dest() : op.src();
+
+  auto component = op->getParentOfType<ComponentOp>();
+  Block *body = component.getBody();
+  auto it = llvm::find_if(body->getArguments(),
+                          [&](auto arg) { return arg == value; });
+  BlockArgument blockArg = *it;
```

Comment:
If the arg isn't found, I think this could dereference a null pointer. Should this fail (or assert) if `it` is null?